### PR TITLE
everything: Fix persisting

### LINKED ISSUE #4456: Everything settings & db not persisting
I've installed Everything ( https://github.com/lukesampson/scoop-extras/blob/master/bucket/everything.json ) and when a new version came out, my settings and database were wiped out. Investigating further, I found that "Everything.db" and "Everything.ini" are updated in the apps directory, and these two files in the persist directory are not touched by the software. I've tried uninstalling (and deleting the persist directory) and reinstalling, but the .db and .ini files in the persist directory are not being used.

The json file does call for them to persist:
    "persist": [
        "Everything.db",
        "Everything.ini"
    ],

Does this mean that Everything doesn't support keeping the .ini and .db files in a different location?

### PR DESCRIPTION
**Pull request by erogy**
- Closes #4456

### CODE REVIEW

File: bucket/everything.json [hunk 1]
```diff
@@ -14,16 +14,15 @@
...
+        "ensure \"$persist_dir\" | Out-Null",
+        "if (Test-Path \"$persist_dir\\Everything.ini\") { Copy-Item \"$persist_dir\\Everything.ini\" \"$dir\" }",
+        "else { & \"$dir\\Everything.exe\" -install-config null }",
+        "'Everything.db','Bookmarks.csv' |% {",
+        "   if (Test-Path \"$persist_dir\\$_\") { Copy-Item \"$persist_dir\\$_\" \"$dir\" }",
         "}"
     ],
     "uninstaller": {
-        "script": "if (Test-Path \"$dir\\Bookmarks.csv\") { Copy-Item \"$dir\\Bookmarks.csv\" \"$persist_dir\" -Force }"
+        "script": "'Everything.ini','Everything.db','Bookmarks.csv' |% { Copy-Item \"$dir\\$_\" \"$persist_dir\" -ErrorAction SilentlyContinue -Force }"
```
Comment (Ash258) [inline bucket/everything.json:25]: Did you see any manifest with command aliases usage??
Do you think it is good idea to use aliases in readable scripts/definitions??

```suggestion
        "script": "Get-ChildItem \"$dir\\*\" -Include 'Everything.ini', 'Everything.db', 'Bookmarks.csv' | Copy-Item -Destination \"$persist_dir\" -ErrorAction SilentlyContinue -Force"
```

File: bucket/everything.json [hunk 2]
```diff
@@ -14,16 +14,15 @@
...
-        "if (!(Test-Path \"$persist_dir\\Everything.ini\")) { & \"$dir\\Everything.exe\" -install-config null }",
-        "if (!(Test-Path \"$persist_dir\\Everything.db\")) { New-Item \"$dir\\Everything.db\" | Out-Null }",
-        "if (Test-Path \"$persist_dir\\Bookmarks.csv\") {",
-        "    Copy-Item \"$persist_dir\\Bookmarks.csv\" \"$dir\"",
-        "} else {",
-        "    New-Item \"$dir\\Bookmarks.csv\" | Out-Null",
+        "ensure \"$persist_dir\" | Out-Null",
+        "if (Test-Path \"$persist_dir\\Everything.ini\") { Copy-Item \"$persist_dir\\Everything.ini\" \"$dir\" }",
+        "else { & \"$dir\\Everything.exe\" -install-config null }",
+        "'Everything.db','Bookmarks.csv' |% {",
```
Comment (Ash258) [inline bucket/everything.json:20]: Comment above

File: bucket/everything.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "version": "1.4.1.987",
-    "description": "Locate files and folders by name instantly.",
+    "description": "Locate files and folders by name instantly",
```
Comment (Ash258) [inline bucket/everything.json:3]: Why do you fix something which is not broken??

```suggestion
    "description": "Locate files and folders by name instantly.",
```

image

Comment (erogy) [reply bucket/everything.json:3]: Because in one of my previous PRs you literally just suggested to remove the dot at the end of description.

Comment (Ash258) [reply bucket/everything.json:3]: Because it was not valid English sentence.

File: bucket/everything.json (hunk 1 above)
Comment (erogy) [reply bucket/everything.json:25]: While I see your point I don't think it's a bad idea either.

File: bucket/everything.json [hunk 4]
```diff
@@ -14,16 +14,13 @@
...
-        "if (!(Test-Path \"$persist_dir\\Everything.db\")) { New-Item \"$dir\\Everything.db\" | Out-Null }",
-        "if (Test-Path \"$persist_dir\\Bookmarks.csv\") {",
-        "    Copy-Item \"$persist_dir\\Bookmarks.csv\" \"$dir\"",
-        "} else {",
-        "    New-Item \"$dir\\Bookmarks.csv\" | Out-Null",
-        "}"
+        "ensure \"$persist_dir\" | Out-Null",
+        "if (Test-Path \"$persist_dir\\Everything.ini\") { Copy-Item \"$persist_dir\\Everything.ini\" \"$dir\" }",
+        "else { Invoke-ExternalCommand \"$dir\\Everything.exe\" -Args @('-install-config null') }",
+        "Get-ChildItem \"$persist_dir\\*\" -Include 'Everything.db', 'Bookmarks.csv' | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue"
```
Comment (Ash258) [inline bucket/everything.json:20]:
```suggestion
        "if (!(Test-Path \"$persist_dir\\Everything.ini\")) { Invoke-ExternalCommand \"$dir\\Everything.exe\" -Args @('-install-config null') | Out-Null }",
        "Get-ChildItem \"$persist_dir\\*\" -Include 'Everything.db', 'Bookmarks.csv', 'Everything.ini' | Copy-Item -Destination \"$dir\" -ErrorAction SilentlyContinue"
```

**Comment by redactedscribe** [comment]
I just installed everything-lite, and it is has the same/similar issue: nothing is created in its persist directory, and the .ini and .db are along side the exe in apps.

Since they are essentially the same program, I think the very same fix could be applied to everything-lite.

**Pull request merged**